lib/bcdice/game_system/OrgaRain.rb

```diff
@@ -31,46 +31,38 @@ def initialize(command)
       end
 
       register_prefix(
-        '(\d+)?OR([0-9])?([0-9])?([0-9])?([0-9])?([0-9])?([0-9])?'
+        '(\d+)?OR(\d{0,6})?'
       )
 
       def eval_game_system_specific_command(command)
-        if command =~ /(\d+)?OR([0-9])?([0-9])?([0-9])?([0-9])?([0-9])?([0-9])?$/i
-          diceCount = (Regexp.last_match(1) || 1).to_i
-          countNo = [(Regexp.last_match(2) || -1).to_i, (Regexp.last_match(3) || -1).to_i, (Regexp.last_match(4) || -1).to_i, (Regexp.last_match(5) || -1).to_i, (Regexp.last_match(6) || -1).to_i, (Regexp.last_match(7) || -1).to_i]
-          countNo.delete(-1)
-          countNo = countNo.sort
+        m = command.match(/(\d+)?OR(\d{0,6})$/i)
+        return nil unless m
 
-          return checkRoll(diceCount, countNo)
-        end
-
-        return nil
+        dice_count = (m[1] || 1).to_i
+        count_no = (m[2] || "").split(//).map(&:to_i).sort
```

Comment:
`split(//)` を `each_char` にすると、意図が分かりやすいです。

```suggestion
        count_no = (m[2] || "").each_char.map(&:to_i).sort
```